test/Microsoft.AspNetCore.SignalR.Client.Tests/HubConnectionTests.cs

```diff
@@ -119,6 +120,33 @@ public async Task ConnectionTerminatedIfServerTimeoutIntervalElapsesWithNoMessag
             Assert.Equal($"Server timeout ({hubConnection.ServerTimeout.TotalMilliseconds:0.00}ms) elapsed without receiving a message from the server.", exception.Message);
         }
 
+        [Fact]
+        public async Task InherentKeepAliveFeatureCanPreventServerTimeout()
+        {
+            using (StartVerifiableLog(out var loggerFactory))
+            {
+                var testConnection = new TestConnection();
+                testConnection.Features.Set<IConnectionInherentKeepAliveFeature>(new TestKeepAliveFeature() { HasInherentKeepAlive = true });
+                var hubConnection = CreateHubConnection(testConnection, loggerFactory: loggerFactory);
+                hubConnection.ServerTimeout = TimeSpan.FromMilliseconds(1);
+
+                await hubConnection.StartAsync().OrTimeout();
+
+                var closeTcs = new TaskCompletionSource<Exception>();
+                hubConnection.Closed += ex =>
+                {
+                    closeTcs.TrySetResult(ex);
+                    return Task.CompletedTask;
+                };
+
+                await Task.Delay(1000);
```

Comment:
For flaky test avoidance reasons, maybe "fake" this passage of time by manually decreasing `_nextActivationServerTimeout` by 1000 ms.

Reply:
It's still using a `Timer` internally, which has a ~15ms minimum delay already, so setting the `ServerTimeout` to 1ms should guarantee that if the timer runs then the server timeout code should run.

Reply:
But that gives me an idea, I could wait for a ping to be sent out by the client which means that the timer ran and therefore that the server timeout ran

Reply:
Instead of doing this, can we do what we do on the server:

https://github.com/aspnet/SignalR/blob/3d6e1e69fd03d0aed50f713ece482e7df65434a6/test/Microsoft.AspNetCore.Http.Connections.Tests/HttpConnectionDispatcherTests.cs#L451

Call the method that runs per tick to force behavior instead of waiting for the timer?

Reply:
We're using your fancy `TimerAwaitable`, so there is no function we can call
https://github.com/aspnet/SignalR/blob/f0c4170e42e72462c05cf1ac14b820e38731da90/src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs#L899-L901

Reply:
Make a function.